### PR TITLE
Fix tests on GDAL 2.2 and some warnings

### PR DESCRIPTION
**Pull request by lnicola**
This includes #52 and might need to be rebased (although I don't think so).

### CODE REVIEW

File: src/vector/tests/mod.rs [hunk 1]
```diff
@@ -198,8 +198,12 @@ fn test_geom_fields() {
...
 fn test_get_layer_by_name() {
     let mut ds = Dataset::open(fixture!("roads.geojson")).unwrap();
-    let layer = ds.layer_by_name("OGRGeoJSON").unwrap(); // change to "roads" for GDAL >= 2.2
-    assert_eq!(layer.name(), "OGRGeoJSON"); // change to "roads" for GDAL >= 2.2
+    if let Ok(layer) = ds.layer_by_name("OGRGeoJSON") {
+        assert_eq!(layer.name(), "OGRGeoJSON");
+    }
+    if let Ok(layer) = ds.layer_by_name("roads") {
+        assert_eq!(layer.name(), "roads");
+    }
```
Comment (jdroenner) [inline src/vector/tests/mod.rs]: Please add some comments about the two cases.

File: src/vector/tests/mod.rs [hunk 2]
```diff
@@ -159,7 +159,7 @@ fn test_json() {
 fn test_schema() {
     let mut ds = Dataset::open(fixture!("roads.geojson")).unwrap();
     let layer = ds.layer(0).unwrap();
-    assert_eq!(layer.name(), "OGRGeoJSON".to_string()); // change to "roads" for GDAL >= 2.2
+    assert!(layer.name() == "OGRGeoJSON" || layer.name() == "roads");
```
Comment (jdroenner) [inline src/vector/tests/mod.rs]: Please add some comments about the two cases. I guess that someone will get confused by this :)

**Comment by jdroenner** [comment]
Let me now when this is no longer WIP ;) Also please rebase if needed. You cloud also close the other one if it is included in this one...

**Comment by lnicola** [comment]
I'll rebase and finish this tomorrow, so you can merge the other one if it looks fine to you.

**Comment by lnicola** [comment]
Done, I think.

**Comment by jdroenner** [comment]
great! thank you :)

**Pull request merged**